Release/v0.5.0

jasonbahl opened the pull request:
# Release Notes

This release focuses _primarily_ on Location Rules and how ACF Field Groups are mapped to the WPGraphQL Schema. 

## Bugfixes / Chores

- [#174](https://github.com/wp-graphql/wp-graphql-acf/pull/174): Typo fix in README. Thanks @pickleat! 
- [#175](https://github.com/wp-graphql/wp-graphql-acf/pull/175): Fixes problem with return type on Select fields. Thanks @ljanecek!
- [#191](https://github.com/wp-graphql/wp-graphql-acf/pull/191): Replaces deprecated `acf_get_term_post_id()` method. Thanks @sboerrigter!
- [#197](https://github.com/wp-graphql/wp-graphql-acf/pull/197): Adds URI to play nice with Github Plugin Updater
- [#235](https://github.com/wp-graphql/wp-graphql-acf/pull/235): Updates README to include docs for registering field groups in PHP. Thanks @XAce90!

## New Features / Big Changes

**LOCATION, LOCATION, LOCATION!!!**

This release primarily addresses issues related to Field Group location rules and adding field groups to the GraphQL Schema. 

Prior to this release, ACF Field Groups were added to the GraphQL Schema strictly by analyzing the ACF Location Rules for the field group(s) and attempting to map the field group to the Schema. 

Some rules are quite nuanced and hard to translate, and in some cases this meant that the Field Group would simply not show up in the Schema, or in other cases the field group wouldn't show exactly where you wanted. 

This release brings a new way of mapping ACF Field Groups to the WPGraphQL Schema. 

It still uses ACF Location Rules to try and "guess" where in the Schema the field group should show, but now it shows you where it will be and allows you to opt-out of the auto-mapping and set the GraphQL Types the field group should show on manually. 

🎥   👉   This video walks through this new functionality: https://youtu.be/VvrZGrcwv0Y

### Closes the following issues:

- closes https://github.com/wp-graphql/wp-graphql-acf/issues/58
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/149
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/169
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/206
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/76
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/135
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/198
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/189
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/186
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/188
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/210
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/22
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/42
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/52
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/65
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/127
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/132
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/162
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/203
- closes https://github.com/wp-graphql/wp-graphql-acf/issues/233

----

📣  **HUGE** shout out to @drewbaker @rsm0128 and @funkhaus for putting a lot of time and energy into Location Rule revamp. Would not be where it is without your work!!! 🙌🏻

pickleat committed:
Update README.md

found typo in line 1738. "assinged" => "assigned"

ljanecek committed:
Fix problem with return value of select format

This is the fix of this issue: https://github.com/wp-graphql/wp-graphql-acf/issues/44

sboerrigter committed:
Replace depricated function acf_get_term_post_id() with term_ string

dgaidula committed:
fix plug-in URI to support github plugin updater

rsm0128 committed:
Fix #135. Add Explicit options for adding ACF Field Groups to the Schema

rsm0128 committed:
Fixed formatting issues. Reverted phpcs auto formatting error and followed WP convention for new code

rsm0128 committed:
fixed spelling issue. Thanks @drewbaker

jasonbahl committed:
- fix the Github Plugin URI

jasonbahl committed:
Merge pull request #214 from jasonbahl/bug/#213-fix-github-plugin-uri

#213 - fix the Github Plugin URI

jasonbahl committed:
Merge branch 'develop' into patch-1

jasonbahl committed:
Merge pull request #197 from dgaidula/patch-1

fix plug-in URI to support github plugin updater

jasonbahl committed:
Merge pull request #191 from sboerrigter/develop

Replace depricated acf_get_term_post_id() function

jasonbahl committed:
Merge pull request #175 from ljanecek/patch-1

Fix problem with return value of select format

jasonbahl committed:
Merge pull request #174 from pickleat/patch-1

Update README.md

jasonbahl committed:
- updates the WPGraphQL for ACF Test environment to properly run tests in Github actions when Pull Requests are opened.
- Updates test for range field to assert it's a float

jasonbahl committed:
Merge pull request #218 from jasonbahl/feature/update-test-env

Update CI Environment for tests

jasonbahl committed:
- update composer.json
- update app.entrypoint
- update Dockerfile to install zip/unzip and pcov

jasonbahl committed:
- update composer.json
- update testing.Dockerfile

jasonbahl committed:
- update github action to install all dependencies

jasonbahl committed:
- add zip extension to PHP setup

jasonbahl committed:
- add zip extension to PHP setup

jasonbahl committed:
- add zip extension to PHP setup

jasonbahl committed:
- add coveralls.yml config

jasonbahl committed:
remove currently unused code quality check from Github workflows

jasonbahl committed:
Merge pull request #220 from jasonbahl/feature/ci-tests

Update test environment

markkelnar committed:
Add json export example file for field groups

markkelnar committed:
Update some examples and image links

jasonbahl committed:
Merge pull request #227 from markkelnar/docs/example-json

Readme and the example json

rsm0128 committed:
updated unit test with new graphql_types_on field

rsm0128 committed:
Aded Explicit Options WpUnit test case

rsm0128 committed:
updated plugin version check

rsm0128 committed:
fixed typo issue with user register fields

rsm0128 committed:
Added contentNode and termNode support for the explicit option

jasonbahl committed:
- add plaftform_check to .gitignore
- update Docker entrypoint to download wp-graphql from WordPress.org
- update test for custom fields

markkelnar committed:
Enable xdebug when USING_XDEBUG specified

markkelnar committed:
Increase php upload filesize

markkelnar committed:
Add development documentation steps

jasonbahl committed:
- Update GraphQL Settings fields to use the GraphQL Type names as values instead of prefixing with `post_type__`, `taxonomy__`, etc
- Update to use `graphql_types` instead of `graphql_types_on` setting name
- Update Checkbox markup to be re-usable for multiple interfaces and their possible types
- Update jQuery for checkbox state management to be re-usable for any Interface checkbox set

jasonbahl committed:
- Add the `ContentTemplate` interface to the Checkboxes

jasonbahl committed:
- Fix typo in JS for unsetting checked state

markkelnar committed:
Download tagged release version of wp-graphql zip

markkelnar committed:
Use .env for SUITES value

markkelnar committed:
Use the .env file, for now

markkelnar committed:
Download wp-graphql.zip then install because of weird directory name thing that was happening

jasonbahl committed:
- add filter to `ContentTemplate` fields to pass the source node along as context
- Add ACF Options Pages to the WPGraphQL when the Schema is loading
- Update code to use `graphql_types` instead of `graphql_types_on`
- Update tests to use `graphql_types` instead of `graphql_types_on`
- run `composer install --no-dev`

markkelnar committed:
Get latest wp-graphql from .org

markkelnar committed:
Support wp-graphql core version download

markkelnar committed:
Variable for base plugin is now WPGRAPHQL_VERSION

jasonbahl committed:
- removes activation script that tries to upgrade location rules
- Update default `required` value for the `graphql_field_name` rule
- Updates docblocks in class-config
- Updates class-config to get location rules from the location rules class
- Adds new LegacyLocationRules class to help support legacy location rules (before GraphQL Schema Types were explicitly set for each field group)
- Adds JS to handle required field logic for graphql_field_name
- Adds tests for CommentLocation, LegacyRules, MenuItemLocation, MenuLocation, TemplateLocation and TermLocation
- updates composer autoloader map

jasonbahl committed:
- Update test to include nav menu registration

jasonbahl committed:
- Update test message linking to WPGraphQL Issue

markkelnar committed:
Add -c build option to build without cache

markkelnar committed:
Something went wrong text

jasonbahl committed:
- Update ACF Settings tp include an AJAX action/callback for determining field group location rules
- Move GraphQL Settings into their own meta box
- Register an `AcfFieldGroup` Interface
- Revamp location-rules. They will now be treated as more of a first-class citizen than a legacy thing. Field Groups will still use ACF Location Rules as the first attempt to map field groups to the Schema, but still allow users to override if they choose to.
- Work on JS for conditional visibility of GraphQL Settings fields
- Update LocationRulesTest (instead of LegacyLocationRulesTest)

jasonbahl committed:
Merge pull request #244 from markkelnar/fix/using-xdebug

Update to xdebug 3 config and add ability to build app docker with xdebug

jasonbahl committed:
Merge commit '3bf411f270c591a89dce3b41acb655046139a10e' into bugfix-135

# Conflicts:
#	docker/app.entrypoint.sh

jasonbahl committed:
- Move GraphQL Field Group Settings into their own box in the ACF Field Group page
- Add Ajax callback for determining rules
- Update Location Rules to check for rule conflicts, and support many edge cases, such as assigning a Field Group to a post_status
- Update JS helpers for the GraphQL Field Group settings. Add conditional logic to the field that allows users to manually chose GraphQL fields or map based on location rules
- Reset location rules when manual graphql type mapping is unchecked.

jasonbahl committed:
- remove dummy code from wp-graphql-acf
- add support for new location rules to map to options pages
- check to see if field group is set to be manually mapped to GraphQL Types, and if so, don't use get_location_rules() to automatically map when building the Schema
- Add test for ACF settings pages and ACF sub settings pages

jasonbahl committed:
- update testing matrix

jasonbahl committed:
- update testing matrix

jasonbahl committed:
- Update tests to test with multiple versions of WPGraphQL

jasonbahl committed:
- Update .gitignore to not ignore the platform check
- back out the change to remove the prefix from field groups (will be addressed later)

jasonbahl committed:
Merge pull request #229 from rsm0128/bugfix-135

Fix #135. Add Explicit options for adding ACF Field Groups to the Schema

jasonbahl committed:
- Adjust the JS on the GraphQL Settings to only execute the AJAX request if one is not already pending.

markkelnar commented on readme.txt:
```diff
@@ -4,7 +4,7 @@ Donate link: https://wpgraphql.com/acf
 Requires at least: 5.0
 Tested up to: 5.1.1
-Stable tag: 0.4.1
+Stable tag: 0.5.0
```
Since this is significant breaking changes, should you put something in the readme description or add an  'upgrade notice' section? See https://wordpress.org/plugins/readme.txt

jasonbahl replied:
Since the plugin isn't distributed on WordPress.org (yet) I've been handling all my communication in Github releases. 

Once we hit 1.0 and distribute on the .org repo, we'll use this README much more actively like I'm doing for WPGraphQL 

This shouldn't actually cause breaking changes either. It's for sure possible, but the tests we did have show that the location rules that used to work still work, and added more coverage for additional location rules. 

I bumped it from 0.4.1 to 0.5.0 because it is significant changes, but should still be non-breaking.

markkelnar commented on src/location-rules.php:
```diff
@@ -0,0 +1,977 @@
+		foreach ( $this->unset_types as $field_group => $types ) {
+			if ( ! empty( $types ) ) {
+				foreach ( $types as $type ) {
+					if ( isset( $this->mapped_field_groups[ $field_group ] ) ) {
```
Looks like this if isset `mapped_field_groups[ $field_group ]` could be moved outside the foreach ($types).

markkelnar replied:
And should the $this->mapped_field_groups be $mapped_field_groups for this is isset check?

jasonbahl replied:
@markkelnar super helpful feedback! 

I just pushed up some changes to simplify this and stepping through caused me to find a small bug in the JS as well! 

Super cool. Thanks for taking the time to go through this hot mess of a PR 😄

jasonbahl committed:
- Update AJAX callback to get the rules as just values not associative array
- Update get_rules() method to early return quicker
- Update main.js to trigger the change after check/uncheck has occurred, not only when checking has occurred
